src/Tasks.UnitTests/AssemblyDependency/ResolveAssemblyReferenceEnvironmentVirtualizationTests.cs

```diff
@@ -0,0 +1,190 @@
+
+            Directory.CreateDirectory(testPath);
+            try
+            {
+                rarTask.Assemblies = new ITaskItem[] { new TaskItem(@"..\RawFileNameRelative\System.Xml.dll") };
+                rarTask.SearchPaths = new string[] { "{RawFileName}" };
+
+                TaskExecutionContext taskExecutionContext = new TaskExecutionContext(testPath, null, null, null);
+                (rarTask as IConcurrentTask).ConfigureForConcurrentExecution(taskExecutionContext);
+
+                Execute(rarTask);
+
+                Assert.Single(rarTask.ResolvedFiles);
+                Assert.Equal(Path.Combine(testPath, "System.Xml.dll"), rarTask.ResolvedFiles[0].ItemSpec);
+            }
+            finally
+            {
+                if (Directory.Exists(testPath))
+                {
+                    FileUtilities.DeleteWithoutTrailingBackslash(testPath);
+                }
+            }
+        }
+
+
+        /// <summary>
+        /// If a relative searchPath is passed in through the search path parameter
+        /// then try to resolve the file but make sure it is a full name with current directory virtualization.
+        /// </summary>
+        [Fact]
+        [Trait("Category", "mono-osx-failing")]
+        public void RelativeDirectoryResolverWithActiveDirectoryVirtualization()
+        {
+            ResolveAssemblyReference rarTask = new ResolveAssemblyReference();
+
+            rarTask.BuildEngine = new MockEngine(_output);
+
+            string testPath = Path.Combine(Path.GetTempPath(), @"RawFileNameRelative");
+            string previousCurrentDirectory = Directory.GetCurrentDirectory();
+
+            Directory.CreateDirectory(testPath);
+            try
+            {
+                rarTask.Assemblies = new ITaskItem[] { new TaskItem(@"System.Xml.dll") };
+                rarTask.SearchPaths = new string[] { "..\\RawFileNameRelative" };
+
+                TaskExecutionContext taskExecutionContext = new TaskExecutionContext(testPath, null, null, null);
+                (rarTask as IConcurrentTask).ConfigureForConcurrentExecution(taskExecutionContext);
+
+                Execute(rarTask);
+
+                Assert.Single(rarTask.ResolvedFiles);
+                Assert.Equal(Path.Combine(testPath, "System.Xml.dll"), rarTask.ResolvedFiles[0].ItemSpec);
+            }
+            finally
+            {
+                if (Directory.Exists(testPath))
+                {
+                    FileUtilities.DeleteWithoutTrailingBackslash(testPath);
+                }
+            }
+        }
+
+
+        /// <summary>
+        /// If a relative file name is passed in through the HintPath then try to resolve directly to that file name and make it a full path with current directory virtualization.
+        /// </summary>
+        [Fact]
+        [Trait("Category", "mono-osx-failing")]
+        [Trait("Category", "netcore-osx-failing")]
+        [Trait("Category", "netcore-linux-failing")]
```

Comment:
There is a problem I recently discovered with unit tests for RAR in unix. ResolveAssemblyReferenceTestFixture does not work correctly with unix paths. A lot of tests currently are just excluded for unix/mac because of that (well, that is my suggestion). I managed to fix ResolveAssemblyReferenceTestFixture for all other tests I introduced except this one. This one (and other tests that are currently just do not run in unix) need much more fixes in ResolveAssemblyReferenceTestFixture. I think we should fix it in another PR.

Reply:
so, the problem is in the testing system, with file system's mocking, not with RAR.

Reply:
hmm, so far I saw the only problem with this class which prevents it to be cross-plat: the comparison of paths (well, at least tests I saw were failing because of that). Yes, this is a big class and you need to fix it in a lot of places, that's why I did not want to do this in this PR. These is very technical fix and I did not think it will take so much time. But maybe you are right, there also might be other changes needed to make it cross platform, I did not thought much about that.